### PR TITLE
fix(tracing): split logging from tracing

### PR DESCRIPTION
**Pull request by jacobmarble**
Helps #1012 

This change splits logging and tracing, so they are configured and
output to distinct destinations. Adds support for writing traces to
an OTLP destination, in addition to Jaeger, stdout, and stderr.

The "rust" and "logfmt" logging formats both changed.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Comment by tustvold** [comment]
I'm not sure tokio-tracing and envlogger are interchangeable. `tokio-tracing` allows for logging with spans included in the logs, which is incredibly useful for debugging highly concurrent systems where log lines may be interleaved. FWIW the tokio-tracing envfilter is additionally more flexible, allowing enabling logs within spans or based on span metadata.

File: Cargo.toml [hunk 1]
```diff
@@ -89,10 +89,10 @@ tokio-stream = { version = "0.1.2", features = ["net"] }
 tokio-util = { version = "0.6.3" }
 tonic = "0.4.0"
 tonic-health = "0.3.0"
-tracing = { version = "0.1", features = ["release_max_level_debug"] }
-tracing-futures = "0.2.4"
```
Comment (domodwyer) [inline Cargo.toml:93]: I am pretty sure we need this to generate the [correct spans on async](https://github.com/tokio-rs/tracing#in-asynchronous-code) methods.

**Review by domodwyer — COMMENTED** [review]
I'm also not sure if this is the best path - using a logger that has no integration with the tracing throws a lot of useful information away (including a fair bit of "free" stuff provided by the 'tracing ecosystem'). 

I'm sure you've considered the tradeoffs before doing this - what's your reasoning for the switch?

**Comment by jacobmarble** [comment]
> I'm not sure tokio-tracing and envlogger are interchangeable. tokio-tracing allows for logging with spans included in the logs, which is incredibly useful for debugging highly concurrent systems where log lines may be interleaved. FWIW the tokio-tracing envfilter is additionally more flexible, allowing enabling logs within spans or based on span metadata.

> I'm also not sure if this is the best path - using a logger that has no integration with the tracing throws a lot of useful information away (including a fair bit of "free" stuff provided by the 'tracing ecosystem').
> 
> I'm sure you've considered the tradeoffs before doing this - what's your reasoning for the switch?

There are certainly benefits to trace logs and to standard logs. I mean to unmix the two in this change. Trace logs should provide deep insight into a sampled subset (0% ~ 100%) of operations, while standard logs should be filtered by level. Additionally, logs and traces are ingested by separate systems - Fluentd->Greylog and Jaeger->Elastic.

For me, the confusion is in the common interface between the `log` crate macros and the `tracing` crate macros. It might be best if we only import `log` and `tracing` to clarify that `tracing::info()` is not `log::info()`. However, it appears that the `tracing` crate intends to replace `log`, so I'll keep reading before I create strong opinions about this.

https://docs.rs/tracing/0.1.25/tracing/#log-compatibility

File: Cargo.toml (hunk 1 above)
Comment (jacobmarble) [reply Cargo.toml:93]: Fixed.

**Comment by carols10cents** [comment]
> For me, the confusion is in the common interface between the `log` crate macros and the `tracing` crate macros. It might be best if we only import `log` and `tracing` to clarify that `tracing::info()` is not `log::info()`. However, it appears that the `tracing` crate intends to replace `log`, so I'll keep reading before I create strong opinions about this.

One option is that [`tracing` provides *more* macros than `log`](https://docs.rs/tracing/0.1.25/tracing/index.html#macros): in addition to `info!`, for example, there is `info_span!`. I'm not sure yet if this solves all your problems; still reading :)

**Pull request closed without merging**